File: easy/reverse-string/python/reverse_string.py
def reverse_string(s: str) -> str:
    """
    Reverse the given string.

    Args:
        s (str): The input string.

    Returns:
        str: The reversed string.
    """
    return s[::-1]


if __name__ == "__main__":
    test_cases = [
        ("hello", "olleh"),
        ("Hacktoberfest", "tsefrebotkcaH"),
        ("Python", "nohtyP"),
        ("a", "a"),
        ("", ""),
    ]

    for text, expected in test_cases:
        result = reverse_string(text)
        assert (
            result == expected
        ), f"Failed for '{text}': expected '{expected}', got '{result}'"
    print("All test cases passed.")

File: easy/reverse-string/cpp/reverse_string.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

string reverseString(const string& s) {
    string result = s;
    reverse(result.begin(), result.end());
    return result;
}

int main() {
    // Test Case 1
    cout << "Test 1: " << reverseString("hello") << " (Expected: olleh)" << endl;
    
    // Test Case 2
    cout << "Test 2: " << reverseString("Hacktoberfest") << " (Expected: tsefrebotkcaH)" << endl;
    
    // Test Case 3
    cout << "Test 3: " << reverseString("Python") << " (Expected: nohtyP)" << endl;
    
    // Test Case 4
    cout << "Test 4: " << reverseString("a") << " (Expected: a)" << endl;
    
    // Test Case 5
    cout << "Test 5: " << reverseString("") << " (Expected: )" << endl;
    
    return 0;
}
